include/psa/crypto_sizes.h

```diff
@@ -1086,4 +1086,33 @@
 #define PSA_CIPHER_FINISH_OUTPUT_MAX_SIZE           \
     (PSA_BLOCK_CIPHER_BLOCK_MAX_SIZE)
 
+/** A sufficient output buffer size for psa_pake_output().
+ *
+ * If the size of the ciphertext buffer is at least this large, it is
+ * guaranteed that psa_pake_output() will not fail due to an
+ * insufficient ciphertext buffer size. The actual size of the output might be
+ * smaller in any given call.
+ *
+ * See also #PSA_PAKE_OUTPUT_MAX_SIZE
+ *
+ * \param alg           A PAKE algorithm (PSA_ALG_XXX value such that
+ *                      #PSA_ALG_IS_PAKE(\p alg) is true).
+ * \param cipher_suite  A cipher suite that is compatible with algorithm \p alg.
+ * \param output        An output type used with algorithm \p alg.
+ * \return              A sufficient output buffer size for the specified
+ *                      output, cipher suite and algorithm. If the cipher suite,
+ *                      the output type or PAKE algorithm is not recognized, or
+ *                      the parameters are incompatible, return 0.
+ */
+#define PSA_PAKE_OUTPUT_SIZE(alg, cipher_suite, output)
```

Comment:
I hadn't thought of this, but now that `cipher_suite` is a structure, it can't easily be a compile-time constant anymore. Can we still have compile-time-constant cipher suite structures such that a call to this macro can evaluate to a compile-time constant?

Reply:
Structures can't be compile-time constants - so the above macro cannot generate a compile-time constant used to dimension an array at compile time.

The cipher-suite needs to be encoded into a scalar (ideally integral) value/values to be usable as a compile-time constant.

One approach would be to not configure the output size on the cipher suite, but just provide a sufficient size based on the algorithm alone. This could be 'close enough' to avoid the complexity of runtime allocation.

An extension of that would be to have `PSA_PAKE_OUTPUT_SIZE()` take the most significant elements (as regards impact on output size) of the cipher_suite as parameters, and provide macro names to directly provide those elements for the common/standard cipher_suites.

Another approach would be to have the common PAKE cipher suites given a 'PAKE ID' value and use that as a shortcut for applications that want to use a standard construction - which would enable compile-time construction of right-sized output buffers (rather than using a MAX_SIZE buffer). Applications using the built-it-yourself cipher suite structure would have to either use dynamic allocation (via a non-constant `psa_pake_output_size()` function) or go for a MAX_SIZE buffer.

Reply:
How about going back to having a psa_pake_primitive_t member replacing the type, family and bits trio in the ciphersuite?

Then this macro could take a primitive and still provide a useful size in compile time.

From the PSA Spec point of view:
- this type would be opaque, but with the restriction that implementations must realise it as an integral type.
- the spec would still need to define how the user can set a specific primitive, but the constructor macro would translate between the standardised representation and the possibly packed internal representation

What do you think?